### PR TITLE
Lps 95418 password service

### PR DESCRIPTION
**Pull request by arthurchan35**
@csierra 
@martamedio 

Hello,

Sorry I should have figured it out eailier, that I was using an older branch.
Anyway, here it is the latest code with SF that came recently.

This pull contains:
1. The libarary
2. Password service - part of Password Management

This one should have no ci failure(test locally, and hopefuly the same on ci)

### CODE REVIEW

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:sf

File: modules/apps/portal-crypto-hash-generator/portal-crypto-hash-generator-message-digest/src/main/java/com/liferay/portal/crypto/hash/generator/message/digest/MessageDigestHashGenerator.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash-generator/portal-crypto-hash-generator-message-digest/src/main/java/com/liferay/portal/crypto/hash/generator/message/digest/MessageDigestHashGenerator.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash-generator/portal-crypto-hash-generator-message-digest/src/main/java/com/liferay/portal/crypto/hash/generator/message/digest/factory/MessageDigestHashGeneratorFactory.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash-generator/portal-crypto-hash-generator-message-digest/src/main/java/com/liferay/portal/crypto/hash/generator/message/digest/factory/MessageDigestHashGeneratorFactory.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/flavor/HashFlavorImpl.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/flavor/HashFlavorImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/context/HashGenerationContextImpl.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/context/HashGenerationContextImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/response/HashGenerationResponseImpl.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/response/HashGenerationResponseImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/processor/HashProcessorImpl.java [hunk 6]
```diff
@@ -0,0 +1,244 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/processor/HashProcessorImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/verification/context/HashVerificationContextImpl.java [hunk 7]
```diff
@@ -0,0 +1,160 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/verification/context/HashVerificationContextImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by arthurchan35** [comment]
#192

**Pull request closed without merging**